### DISCUSSION TITLE
WS Identifier

### DISCUSSION THREAD

**Discussion by atel0** [uNetworking/uWebSockets.js#86] [Q&A] [answered]
Can u add WS IDs, but if i subsribe to message event, i can't identifier what client push this message, i have only ws connect, but what the client send message idk.

P.S. Thx for u project.

**Comment by mrlika** [comment]
You can store data in WebSocket object. WebSocket object is preserved between messages.  See example:

https://github.com/Novage/wt-tracker/blob/master/lib/uws-tracker.ts#L126

**Comment by atel0** [comment]
hm.... 
I thought that this will not work, well, well then, thanks)

**Comment by PeterHewat** [comment, marked as answer]
Hi, this is what I do:
```
  /* ... */
}).ws('/io', {
  open: (ws, req) => {
      ws.uuid = create_UUID();
  /* ... */
// UUID generator from: https://gist.github.com/LeverOne/1308368
function create_UUID(a, b) { for (b = a = ''; a++ < 36; b += a * 51 & 52 ? (a ^ 15 ? 8 ^ Math.random() * (a ^ 20 ? 16 : 4) : 4).toString(16) : '-'); return b }
```
Hope this helps.

**Comment by caracal7** [comment]
Hi, @PeterHewat.

This way works perfect but can I share `uuid` between `.ws()` and other endpoints?

```

UWebSockets.App({})
    .ws('/*', {
        compression: 0,
        maxPayloadLength: 16 * 1024 * 1024,
        idleTimeout: 10,
        open: (ws, req) => {
            ws.uuid = uniqueID();
        }
     })
    .get('/myroute', (res, req) => {
        // uuid ???????    <=======================
    })
    .listen(port, token => {});

```
